hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeInfo.java

```diff
@@ -69,6 +77,24 @@ public void updateLastHeartbeatTime() {
     }
   }
 
+  /**
+   * Updates the last LayoutVersion.
+   */
+  public void updateLastKnownLayoutVersion(LayoutVersionProto version) {
```

Comment:
it is doable. The reason I kept layout version info and its handling separate from the rest of the heartbeat processing is because I wanted to play with the idea that later on, we do not need to send layout version to SCM all the time but only when needed.